core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/samModels.scala

```diff
@@ -262,8 +263,68 @@ object PrivateAzureStorageAccountAction {
     sealerate.collect[PrivateAzureStorageAccountAction].map(a => (a.asString, a)).toMap
 }
 
-// TODO [IA-4608] merge with SamPolicyName
-/** Represents a role in Sam, permitting a set of actions on a resource of a certain type. */
+sealed trait RuntimeRole extends Product with Serializable {
```

Comment:
This is great! Would it save headaches down the line to have these all extend a base `SamResourceRole` for simpler type signatures?

Reply:
That's a good idea, I can do that. Although I think I would like to leave `SamPolicyData` in terms of String for now -- since it's used in `HttpSamDAO`, and I'm not keen to introduce new encoders/decoders.

Reply:
pushed!